core/src/main/java/org/bitcoinj/base/internal/ByteUtils.java

```diff
@@ -221,34 +221,26 @@ public static void writeInt32LE(long val, byte[] out, int offset) throws ArrayIn
 
     /**
      * Write a 32-bit integer to a given buffer in big-endian format.
-     * <p>
-     * The value is expected as an unsigned {@code long} as per the Java Unsigned Integer API.
      *
      * @param val value to be written
      * @param buf buffer to be written into
      * @return the buffer
      * @throws BufferOverflowException if the value doesn't fit the remaining buffer
      */
-    public static ByteBuffer writeInt32BE(long val, ByteBuffer buf) throws BufferOverflowException {
-        checkArgument(val >= 0 && val <= MAX_UNSIGNED_INTEGER, () ->
-                "value out of range: " + val);
+    public static ByteBuffer writeInt32BE(int val, ByteBuffer buf) throws BufferOverflowException {
         return buf.order(ByteOrder.BIG_ENDIAN).putInt((int) val);
     }
 
     /**
      * Write a 32-bit integer to a given byte array in big-endian format, starting at a given offset.
-     * <p>
-     * The value is expected as an unsigned {@code long} as per the Java Unsigned Integer API.
      *
      * @param val    value to be written
      * @param out    buffer to be written into
      * @param offset offset into the buffer
      * @throws ArrayIndexOutOfBoundsException if offset points outside of the buffer, or
      *                                        if the value doesn't fit the remaining buffer
      */
-    public static void writeInt32BE(long val, byte[] out, int offset) throws ArrayIndexOutOfBoundsException {
-        check(offset >= 0 && offset <= out.length - 4, () ->
```

Comment:
Please don't remove these checks and comments.

Reply:
ded back and force-pushed.